monad-stack/src/main/java/com/github/jorgecastillo/kotlinandroid/data/datasource/remote/MarvelNetworkDataSource.kt

```diff
@@ -25,24 +25,26 @@ import kotlinx.coroutines.experimental.async
  * required execution context.
  */
 
-fun fetchAllHeroes() = Reader.ask<IdHK, GetHeroesContext>().map({ ctx ->
+fun fetchAllHeroes() = Reader.ask<ForId, GetHeroesContext>().map({ ctx ->
   IO.monadSuspend().bindingCatch {
-    runInAsyncContext(
+    val result = runInAsyncContext(
         f = { queryForHeroes(ctx) },
         onError = { IO.raiseError<List<CharacterDto>>(it) },
         onSuccess = { IO.pure(it) },
         AC = ctx.threading
     ).bind()
+    result.bind()
```

Comment:
double bind again? Is this intended for any reason? the function results are already binded  the line before.

Reply:
https://github.com/JorgeCastilloPrz/ArrowAndroidSamples/pull/18#discussion_r167400008